drivers/gpio/Kconfig.stm32

```diff
@@ -6,6 +6,7 @@
 menuconfig GPIO_STM32
 	bool "GPIO Driver for STM32 family of MCUs"
 	depends on SOC_FAMILY_STM32
+	select USE_STM32_LL_GPIO
```

Comment:
Shouldn't we namespace these `USE` options properly? So `STM32_USE_LL_GPIO` instead of `USE_STM32_LL_GPIO`.

Reply:
@jukkar that's a good question. I suppose current naming can be changed, even though the purpose of this RFC is to improve how STM32 LL HAL headers are included.

Reply:
IMHO the kconfig options should have some kind of namespacing, and in this case the namespace should probably be `STM32` as this option is related to that soc. The `USE_STM32...` looks like an option that might be used by an application in its own kconfig.

Reply:
cc: @erwango

Reply:
I'm open to change this, but I think this is a separate issue.
Also this is not a simple change as I know these symbols have users in  out-of-tree apps. So we would need to notify this change. I propose to focus on current proposal and have a dedicated issue open on this topic.